includes/migrate.form.inc

```diff
@@ -76,7 +76,13 @@ function islandora_basic_collection_migrate_item_form_submit($form, &$form_state
   $new_collection = islandora_object_load($form_state['values']['new_collection_name']);
   $current_parents = islandora_basic_collection_get_parent_pids($object);
   if ($object && $new_collection) {
-    foreach ($current_parents as $parent) {
+    if ($object === $new_collection) {
+      drupal_set_message(t('Migration aborted. A collection may not be a child of itself.'), 'error');
+      RETURN;
+    }
+
+    foreach ($current_parents as $parents) {
+      $parent = islandora_object_load($parents);
```

Comment:
Also you can remove this, all you are doing is loading the PID into an object and then in `islandora_basic_collection_remove_from_collection()` we grab the [object's id](https://github.com/Islandora/islandora_solution_pack_collection/blob/01d6503a689cb2ad0a87a67d6a9f6af069ec17cb/includes/utilities.inc#L201-L203) anyways.

Reply:
Are you referring specifically to line 85? What happens if it has multiple parents, then?

Reply:
I'm talking specifically about you using `islandora_object_load()` here. This turns the PID (`$parent` or `$parents`) into an object. Then you pass that object to the function below.
```
islandora_basic_collection_remove_from_collection($object, $parent);
```
Then `$parent` is taken into that function as the argument `$collection`:
```
function islandora_basic_collection_remove_from_collection(AbstractObject $member, $collection) {
  if ($collection instanceof AbstractObject) { // Yes it is because we islandora_object_load() it
    $collection = $collection->id; // Turn it back into a PID.
  }
  $member->relationships->remove(FEDORA_RELS_EXT_URI, 'isMemberOfCollection', $collection);
  $member->relationships->remove(FEDORA_RELS_EXT_URI, 'isMemberOf', $collection);
}
```

I think you could probably revert your changes here completely.

Reply:
OK, I get what you mean now. That wasn't actually my code - that ended up in there from resolving a weird merge conflict - but I've taken that out.